flow/src/main/java/com/sequenceiq/flow/service/FlowService.java

```diff
@@ -155,8 +155,10 @@ public FlowCheckResponse getFlowChainStateSafe(Long resourceId, String chainId)
         }
     }
 
-    private void validateResourceId(List<FlowLog> flowlogs, Long resourceId) {
-        if (flowlogs.stream().anyMatch(l -> !l.getResourceId().equals(resourceId))) {
+    private void validateResourceId(List<FlowLog> flowlogs, List<FlowChainLog> relatedChainIds, Long resourceId) {
+        if (relatedChainIds.size() > 0 && relatedChainIds.get(0).getFlowChainType().equals("DatalakeResizeFlowEventChainFactory")) {
```

Comment:
Is it guaranteed that the first item will be the `DatalakeResizeFlowEventChainFactory`?

Reply:
All the entries in FlowChainLog would be associated with the flow chain id provided. Technically, all the entries have the same flow chain type. In case of resizing, it would be DatalakeResizeFlowEventChainFactory.

Reply:
I'm really against it. It is a common flow logic, we should not hardcode here any service-related thing. Anyway, if I am a poweruser in a tenant, then I can query ANY DatalakeResizeFlowEventChainFactory not only for resources in my account. It is a bad idea. I need to think about how we can solve it, but it's not the way.

Reply:
I understand it is not an ideal solution. Will think of another way to solve the issue.

I did not understand the issue you raised with the power user. Can you elaborate?

Reply:
The new solution looks fine to me, thanks!